posit-dev/positron
# Add new 'r.rmarkdownRender' command

### Intent

Addresses #1421 by adding a command to do `rmarkdown::render()` in a fresh R session (i.e. R terminal)

### Approach

This PR adds a command that is similar to `'r.sourceCurrentFile'` but checks to see if rmarkdown is installed and runs it in a new terminal.

### QA Notes

We should check how this command works both with/without the Quarto extension installed. It contributes both a command to the command palette and to the "editor/title/run" bar, but only for `.Rmd` and not `.qmd`. The `'r.sourceCurrentFile'` command should all still work as expected.

You can make a file saved as `.Rmd` such as this:

````markdown
---
title: "A wonderful R Markdown document"
output:
  html_document:
    math_method: katex
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

Here is some math:

$1 + 1$

You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
````
